### ISSUE TITLE
[Bug]: Open quest page command via a function doesn't work on world loading

### ISSUE THREAD

**Issue by Krampus69** [FTBTeam/FTB-Mods-Issues#1422]
### Mod

FTB Quests

### Mod version

ftb-quests-forge-2001.4.9

### NeoForge / Forge / Fabric version

Forge 47.3.12

### Modpack & version

Legendary Edition 1.20.1 (but we didn't release it yet guys)

### What issue are you having?

So, my FTB Quests book has an image that triggers a command, and this command is : 
_"command:function minecraft:open_thirst_page"_

So I made a function in my datapack that has 2 commands triggering one by one, the first one opens the quest page, and the other triggers a local sound I made. Here is what is written in this function :

_"ftbquests open_book 60EC86A4F154048E
playsound ambientsounds:fx.menu.tutorial_pop ambient @s ~ ~ ~ 1"_

The issue is that this doesn't work on world loading. If I click on the image with the command, an error will occur in the chat "UNKNOWN FUNCTION MINECRAFT:OPEN_THIRST_PAGE"
The game can't read the function because the quest ID is not referenced on world loading, so it leads to nothing (c.f the logs) : 

_[14Dec2024 17:41:03.137] [Render thread/ERROR] [net.minecraft.server.ServerFunctionLibrary/]: Failed to load function minecraft:open_temp_page
java.util.concurrent.CompletionException: java.lang.IllegalArgumentException: Whilst parsing command on line 1: No quest file was found, what's going on?!_

!!! IMPORTANT !!! If I reload the datapack with a /reload command, the command works again. So it's not a typo issue, because it works once I reload it. But on world loading, it doesn't. 

Also, another important information : I have others functions in my datapack that I made, they all load except for the ones I made with an "ftbquest open_book" command inside.

You can see the issue in the latest.log file, if you look for "function minecraft" you'll find it easily.

(Thank you for your amazing work btw, love what you do !)

### Crashlogs

https://pste.ch/keculabuya.md

### Steps to reproduce

1. Make a FTB Quest page and copy its ID.
2. Make a datapack with a function containing a "ftbquest open_book" command opening the page you made.
3. Put the datapack in your game, try to call it with /function <name_of_your_datapack_function> to verify it's working.
4. Close the game and open it again
5. Try to call your function (it should not work there)
6. Do a /reload and try to call it again (it should work now!)

**Comment by Krampus69** [comment]
Wait, guys ! I made a video to clarify the issue
https://streamable.com/1cym03

You can see there the button cannot play the command /function minecraft:open_thirst_page at first, but once I reload the datapack it works just fine !

**Comment by desht** [comment]
I did a bit more investigation into this, and there might be a fix?

All functions are parsed at world load time, very early on in the loading process, well before the quest book loads.  If any command doesn't parse, the function doesn't load. Parsing fails here because the `open_book` subcommand tries to get a list of known quests so it can offer a list to the client. But since there's no quest book yet, this fails.

I will experiment with a couple of possible fixes: either allowing a soft-fail if there's no quest book and offering no completions, or if that doesn't work, adding a separate subcommand which just treats the quest ID like a plain string and doesn't try to find it in the quest book at all.

**Comment by desht** [comment]
Fixing commands to work in functions is the right choice here, and I do have this working now in dev, so will be in the next release. I'd rather stick to using MC functions to run sequences of commands than change the click option to take a list. Functions have other advantages like the ability to use macros, for example.

**Comment by Krampus69** [comment]
You're amazing thank you for that 🙏

**Comment by Krampus69** [comment]
Hello ! I see that a fix was commit 3 weeks ago, does it mean it's up in the latest FTB version ?

**Comment by desht** [comment]
Yes, it's available in 2101.1.3 which is out now (note: there will be a 2101.1.4 release today so you might want to wait for that)
